### PR TITLE
import-w3c-tests should display the WPT Upstream commit ID

### PR DESCRIPTION
**Pull request by karlcow**
#### c9b28dc7ce071a1002f715f988f732a471b222ee
<pre>
import-w3c-tests should display the WPT Upstream commit ID
<a href="https://bugs.webkit.org/show_bug.cgi?id=248445">https://bugs.webkit.org/show_bug.cgi?id=248445</a>
rdar://103141678

Reviewed by Jonathan Bedard.

Collect the WPT upstream ID to copy/paste in the import PR
when importing wpt tests into WebKit.

* Tools/Scripts/webkitpy/w3c/test_downloader.py:
(TestDownloader.__init__):
(TestDownloader.checkout_test_repository):
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.__init__):
(TestImporter.do_import):
(TestImporter.import_tests):

Canonical link: <a href="https://commits.webkit.org/269513@main">https://commits.webkit.org/269513@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/79156838aac7e35caca0d7c3ecf4efad53d91457

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22731 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/402 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23818 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24638 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21041 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22990 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1475 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23259 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21962 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/243 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19730 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25491 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/22914 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/240 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20603 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26816 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20613 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20848 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24669 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/277 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18121 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/216 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5430 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/344 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/279 "Built successfully") | | |

### CODE REVIEW

**Comment by karlcow** [comment]
It currently displays something like:

```
------- COPY & PASTE in your PR. -------
WPT Upstream commit ID: 3324586879a38951098d072bf5ae8f56408a9afa
----------------------------------------
```

It can be improved depending on the needs.

**Comment by Ahmad-S792** [comment]
@karlcow - Nice! Such a productivity win... 👍

File: Tools/Scripts/webkitpy/w3c/test_downloader.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def __init__(self, repository_directory, host, options):
         self._test_suites = []
 
         self.repository_directory = repository_directory
+        self.rev_parse_hash = "Missing"
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_downloader.py]:
```suggestion
        self.working_copy_head = None
```

Let's use a more descriptive variable name, and let's not use magic strings.

File: Tools/Scripts/webkitpy/w3c/test_downloader.py [hunk 2]
```diff
@@ -92,6 +93,8 @@ def checkout_test_repository(self, revision, url, directory):
         else:
             git = self.git(directory)
         _log.info('Checking out revision ' + revision)
+        # to be used when logging in test_importer for putting inside a PR
+        self.rev_parse_hash = git.rev_parse('HEAD')
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_downloader.py]: This needs updated for the above suggestion, and also needs to be _after_ the checkout line, because otherwise we're getting the HEAD revision prior to checking out a different revision.

**Review by JonWBedard — APPROVED** [review]

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 3]
```diff
@@ -602,6 +602,10 @@ def import_tests(self):
             self.write_import_log(new_path, copied_files, prefixed_properties, prefixed_property_values)
 
         _log.info('Import complete')
+        _log.info('------- COPY & PASTE in your PR. -------')
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: We shouldn't ask people to copy and paste things into PRs, we should put it into the commit message which automatically gets included as part of the PR via the expected `git-webkit` workflow.

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 4]
```diff
@@ -602,6 +602,10 @@ def import_tests(self):
...
+        wpt_hash_ref = self.test_downloader().rev_parse_hash
+        _log.info('WPT Upstream commit ID: %s', wpt_hash_ref)
```
Comment (nt1m) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]:
```suggestion
        _log.info('Upstream commit: https://github.com/web-platform-tests/wpt/commit/%s', wpt_hash_ref)
```

or:

```
        _log.info('Upstream commit: %s', wpt_hash_ref)
```

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 3 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: @gsnedders 
Yes. I was thinking it would be better, but that doesn't seem possible as it is two completely async and disconnected processes. 

Or maybe there's a better way to do it that I have missed. 

1. open a bug on bugs.webkit.org "Import WPT /css/foobar" -> bugid
2. `git checkout -b bugid-css_foobar`
3. `import-w3c-tests -t css/foobar` (here the moment we can get the WPT commit hash)
4. fix what needs to be.
5. `git webkit commit --issue bugid-css_foobar .`← At this stage the WPT commit hash in the commit message that we edit manually
6. `git webkit pr`

Is there a way to store on the filesystem some information for this in the current process. That would be great if possible.

Comment (gsnedders) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: Ah, true. I was thinking of something unrelated when I wrote that comment.

Instead of "COPY & PASTE in your PR." can we go with something like "Please include the following in your commit message:"?

Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: The message is now 

```
--------- Please include the following in your commit message: ---------

Upstream commit: https://github.com/web-platform-tests/wpt/commit/3d084c0ac239eff15e62958a18b5508203dc2aaa

------------------------------------------------------------------------
```

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 4 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]:
```
Upstream commit: https://github.com/web-platform-tests/wpt/commit/3d084c0ac239eff15e62958a18b5508203dc2aaa
```

File: Tools/Scripts/webkitpy/w3c/test_downloader.py (hunk 2 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_downloader.py]: done.

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 3 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: done.

File: Tools/Scripts/webkitpy/w3c/test_downloader.py (hunk 1 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_downloader.py]: done.

**Comment by karlcow** [comment]
Thanks for the review everyone.

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 5]
```diff
@@ -621,6 +620,12 @@ def import_tests(self):
         for prefixed_value in sorted(total_prefixed_property_values, key=lambda p: total_prefixed_property_values[p]):
             _log.info('  %s: %s', prefixed_value, total_prefixed_property_values[prefixed_value])
 
+        # Fetch the Upstream commit ID for the WPT import So it is ready to insert in the commit message.
+        upstream_id = self.test_downloader().upstream_commitID_head
+        _log.info('\n--------- Please include the following in your commit message: ---------\n')
+        _log.info(f'Upstream commit: https://github.com/web-platform-tests/wpt/commit/{upstream_id}\n')
```
Comment (karlcow) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: Oh the code has to be compatible with Python 2?
@JonWBedard 

```
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/webkitpy/w3c/test_importer.py", line 626
    _log.info(f'Upstream commit: https://github.com/web-platform-tests/wpt/commit/{upstream_id}\n')
                                                                                                 ^
SyntaxError: invalid syntax
```

Comment (gsnedders) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: Yes, currently. We might at some point try to cut down having to support Python 2 throughout webkitpy, but it's potentially not worth the effort versus removing Python 2 support altogether.

File: Tools/Scripts/webkitpy/w3c/test_downloader.py [hunk 6]
```diff
@@ -60,6 +60,7 @@ def __init__(self, repository_directory, host, options):
         self._test_suites = []
 
         self.repository_directory = repository_directory
+        self.upstream_commitID_head = None
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_downloader.py]: this is a weird mix of casing? `upstream_commit_id_head` would make more sense here, but still seems a bit wordy.

that said, I'd rather just go for something shorter like "upstream_revision"?

File: Tools/Scripts/webkitpy/w3c/test_downloader.py [hunk 7]
```diff
@@ -93,6 +94,8 @@ def checkout_test_repository(self, revision, url, directory):
             git = self.git(directory)
         _log.info('Checking out revision ' + revision)
         git.checkout(revision, not self._options.verbose)
+        # to be used when logging in test_importer for putting inside a PR
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_downloader.py]: I don't think this comment adds anything?

File: Tools/Scripts/webkitpy/w3c/test_downloader.py [hunk 8]
```diff
@@ -93,6 +94,8 @@ def checkout_test_repository(self, revision, url, directory):
...
+        self.upstream_commitID_head = git.rev_parse('HEAD')
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_downloader.py]: By only initialising this here, we only have this in the case when `-s` is not passed. Maybe that's good, given the local working copy could have changes in it, but it is perhaps surprising for people who use their own pre-existing WPT clone?

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 9]
```diff
@@ -621,6 +620,12 @@ def import_tests(self):
         for prefixed_value in sorted(total_prefixed_property_values, key=lambda p: total_prefixed_property_values[p]):
             _log.info('  %s: %s', prefixed_value, total_prefixed_property_values[prefixed_value])
 
+        # Fetch the Upstream commit ID for the WPT import So it is ready to insert in the commit message.
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: This has a lot of odd capitalisation, and the comment seems entirely redundant given what the next two lines are?

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 5 above)
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: `upstream_id` be `None`, for example if `-s` is passed. We should handle this being `None` (either by noting the upstream commit is unknown or by not outputting this section at all).

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 10]
```diff
@@ -621,6 +620,12 @@ def import_tests(self):
...
+        upstream_id = self.test_downloader().upstream_commitID_head
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: We call this an "upstream commit" in the output, so why is this an "upstream_id" here?

This means this PR now calls this value three things: "upstream commit", "upstream_id", and "upstrema_commitID_head"—can we just use a single name for this?

File: Tools/Scripts/webkitpy/w3c/test_downloader.py (hunk 8 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_downloader.py]: Let's try something else.

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 9 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: crumbles from prototyping

File: Tools/Scripts/webkitpy/w3c/test_importer.py (hunk 10 above)
Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: yes we can.

File: Tools/Scripts/webkitpy/w3c/test_importer.py [hunk 11]
```diff
@@ -181,6 +181,8 @@ def __init__(self, host, test_paths, options):
         self._potential_test_resource_files = []
 
         self.import_list = []
+        self.upstream_revision = self.test_downloader().upstream_revision
+
```
Comment (gsnedders) [inline Tools/Scripts/webkitpy/w3c/test_importer.py]: This will always be `None` at this point, no? When the downloader hasn't done anything yet, the `upstream_revision` is just `None`, which makes creating the downloader redundant here.

It's also probably more work than we really want in the initialiser anyway? Setting `self.upstream_revision = None` here would be fine.

Also you've added an extra (redundant) new line here.

Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: ooops. I set it twice and I forgot to remove it. 🤦 Thanks for catching that.

Comment (karlcow) [reply Tools/Scripts/webkitpy/w3c/test_importer.py]: done.
